### ISSUE TITLE
Suggestion for Enhanced Input Handling for IME

### ISSUE THREAD

**Issue by AdemJensen** [lmstudio-ai/lmstudio-bug-tracker#75]
When users switch between English and Chinese (or other languages) characters without changing their keyboard layout, they often type in English characters directly through the IME, which temporarily holds these inputs until confirmed. In this context, pressing Enter is necessary to commit the typed text into the chat window rather than select specific characters from an IME candidate list.

However, as currently designed, your software mistakenly interprets the Enter key press as a command to send messages immediately, even when the user intends only to finalize their mixed-language input. This unexpected behavior can lead to unintended message submissions and frustration among users accustomed to this typing style.

To address this issue, I propose an enhancement that would allow your chat software to intelligently differentiate between different stages of text entry based on IME state. Specifically:

- When the user's input method shows a Chinese (or other languages) IME candidate list or is in English input mode through the same IME:

  - Pressing Enter should prioritize processing any temporary buffer from the IME and adding it to the chat window before considering sending the message.

Here is a video to present the issue: I was intended to input "这是一段中文输入PlusE 后续内容", but after I enter "PlusE" I pressed Enter, which just send the partially-done content into the chat zone, I had to switch the keyboard layout to achieve this.

https://github.com/user-attachments/assets/c2420364-b17e-4fbb-95b7-16b4ed93bc3c

And here is a video showing what is intended to be like (I took QQ as an example): when I enter "andE" and press Enter key, the "andE" content just gets confirmed rather than sending the partially-done content.

https://github.com/user-attachments/assets/bffaa3da-b72a-4e85-ac87-cd46f0415c7d

**Comment by morigs** [comment]
I have the same problem with Japanese IME, this unfortunately makes LM Studio chat almost unusable. I have to copy-paste each message from external text editor

**Comment by yagil** [comment]
@AdemJensen and @morigs: in LM Studio 0.3.2 we have an option in Settings to send messages on Shift + Enter, instead of just Enter. Does that solve the problem?

**Comment by morigs** [comment]
@yagil thanks for the tip, it working now!

However, IMO it's better to change the chat event from `onkeyup` to `onsubmit`, I think this should resolve the issue

**Comment by AdemJensen** [comment]
@yagil Yes, working also. Same suggestion as morigs, using `onsubmit` will sole the issue from the source.

**Comment by yagil** [comment]
We fixed this in 0.3.9 build **6**. Please let us know if it works for you!

**Comment by DamianLettie** [comment]
It’s working well. Thank you!Tested v0.3.9 build 6 on macOS v15.3.On 2 Feb 2025, at 12:15, Yagil Burowski ***@***.***> wrote:﻿
We fixed this in 0.3.9 build 6. Please let us know if it works for you!
